File: 剑指Offer/17.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def deleteNode(self, head: ListNode, val: int) -> ListNode:
        index = head
        if head.val == val:
            head = head.next
        while index.next != None and index.next.val != val:
            index = index.next
        
        if index.next != None:
            index.next = index.next.next
        return head

File: 剑指Offer/17.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* deleteNode(ListNode* head, int val) {
        ListNode* index = head;
        if (head->val == val){
            head = head->next;
            // delete index;
            return head;
        }
        while (index->next != NULL && index->next->val != val){
            index = index->next;
        }
        if (index->next != NULL){
            ListNode* temp = index->next;
            index->next = index->next->next;
            delete temp;
        }
        return head;
    }
};